core/server/master/src/main/java/alluxio/master/AlluxioMasterProcess.java

```diff
@@ -149,31 +166,148 @@ public InetSocketAddress getRpcAddress() {
 
   @Override
   public void start() throws Exception {
-    LOG.info("Starting...");
+    LOG.info("Process starting.");
+    mRunning = true;
+    startCommonServices();
```

Comment:
It seems reasonable to start before as some one might want to monitor the metrics during reply.

Though in reality I would rather do minimal work up front, so that on failover things start serving as fast as possible.

For this PR I am fine with being safe and trying to keep things as minimally changed as possible though.